### PR TITLE
Change setSearch to handle SearchParams.t

### PR DESCRIPTION
**Pull request by pedrobslisboa**
## Description

This PR changes the `setSearch` to handle SearchParams.t and add `setSearchString` to handle strings

## New API
```ocaml
let setSearchAsString: (t, string) => t;
let setSearch: (t, SearchParams.t) => t;
```

### CODE REVIEW

File: packages/url/native/URL.re [hunk 1]
```diff
@@ -214,8 +214,12 @@ let search = url => {
   | _ => Some("?" ++ Uri.encoded_of_query(~scheme, query))
   };
 };
-let setSearch = (t, string) => {
-  Uri.with_query(t, Uri.query_of_encoded(string));
+let setSearchString = (t, searchString) => {
```
Comment (davesnx) [inline packages/url/native/URL.re:217]: I would rename this to `setSearchAsString`

**Review by davesnx — APPROVED** [review]
Tiny nit, but overall looks perfect

File: packages/url/native/URL.re [hunk 2]
```diff
@@ -214,8 +214,12 @@ let search = url => {
...
+  Uri.with_query(t, Uri.query_of_encoded(searchString));
+};
+let setSearch = (t, searchParams) => {
+  let queryString = SearchParams.toString(searchParams);
+  setSearchString(t, queryString);
```
Comment (davesnx) [inline packages/url/native/URL.re:222]: no need to transform toString here, Uri.with_query should do the job, right?

Comment (pedrobslisboa) [reply packages/url/native/URL.re:222]: Good point

**Pull request merged**